main.tf

```diff
@@ -1,3 +1,31 @@
-/********************************************************************
-This file is used to implement the ROOT module.
-*********************************************************************/
+module "dynamic_values" {
+  source                            = "./submodules/dynamic_values"
+  enterprise_account_groups         = var.enterprise_account_groups
+  enterprise_accounts               = var.enterprise_accounts
+  enterprise_crn                    = var.enterprise_crn
+  enterprise_primary_contact_iam_id = var.enterprise_primary_contact_iam_id
+}
+
+module "enterprise_hierarchy_depth_0" {
```

Comment:
Now that we have simplified the two inputs from a hierarchy map to a parent reference, do we need to be limited to a depth of 3 anymore? And do we need dynamic_values call at all?

I think if we had two blocks, one for groups that is triggered first, then another block for accounts that has a depends_on for groups, we would not need to limit to hard coded depths anymore and this code would be greatly simplified.

The only thing to figure out would be if terraform will create groups in the proper order (because of implicit dependency of parents), or would this code need to "sort" the group array so that parents get deployed before their children? If we can figure this out you might not need dynamic values or the hierarchy submodule.

Reply:
I cannot think of another way to do this without these depth blocks, so nothing to change here. I'm going to mark this resolved, please ignore.